src/program.js

```diff
@@ -132,6 +133,20 @@ export class Program {
       log.info('Version:', getVersion(absolutePackageDir));
     }
 
+    if (argv.config) {
+      const configObject = loadJSConfigFile(argv.config);
+      try {
+        applyConfigToArgv({
+          argv,
+          configObject,
+          options: this.options,
+          configFileName: argv.config,
+        });
+      } catch (error) {
```

Comment:
@rpl tbh I wrapped it this way because instead of throwing i would get `UnhandledPromiseRejectionWarning` despite `applyConfigToArgv` being a sync function.

Reply:
@saintsebastian that's because the `execute` method is an async function, it should be enough to just move the code that uses `loadJSConfigFile` and `applyConfigToArgv` inside the existent `try...catch` at line 150 of this patch (it was at line 135 before the changes applied by this patch), because it already implements the behavior that we want.

Reply:
@rpl ah! now i see it. thanks! I will add tests and that should probably be it for this patch